pkg/deploy/lattice/service_manager.go

```diff
@@ -2,318 +2,301 @@ package lattice
+	emptyStatus := ServiceStatus{}
 
-			latticeSess := s.cloud.Lattice()
-			resp, err := latticeSess.UpdateServiceWithContext(ctx, &serviceUpdateInput)
-			glog.V(2).Infof("UpdateServiceWithContext >>>> req %v resp %v err %v\n", serviceUpdateInput, resp, err)
+	if svc.Spec.CustomerCertARN != "" {
+		updReq := &UpdateSvcReq{
+			CertificateArn:    aws.String(svc.Spec.CustomerCertARN),
+			ServiceIdentifier: svcSum.Id,
+		}
+		if updReq != nil {
+			_, err := m.cloud.Lattice().UpdateService(updReq)
 			if err != nil {
-				glog.V(6).Infoln("fail to update service")
-				return latticemodel.ServiceStatus{ServiceARN: "", ServiceID: ""}, err
+				return emptyStatus, err
 			}
-
 		}
-
 	}
-	err = s.serviceNetworkAssociationMgr(ctx, service.Spec.ServiceNetworkNames, serviceID)
 
+	err := m.updateSnSvcAssocs(ctx, svc, svcSum)
 	if err != nil {
-		return latticemodel.ServiceStatus{ServiceARN: "", ServiceID: ""}, err
+		return emptyStatus, err
 	}
 
-	return latticemodel.ServiceStatus{ServiceARN: serviceArn, ServiceID: serviceID, ServiceDNS: serviceDNS}, nil
-}
-
-// TODO: Further refactor for all manager under lattice/
-func (s *defaultServiceManager) isServiceNetworkServiceAssociationStatusAssociated(associationStatus string) bool {
-	switch associationStatus {
-	case vpclattice.ServiceNetworkServiceAssociationStatusActive:
-		return true
-	case vpclattice.ServiceNetworkServiceAssociationStatusCreateInProgress:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusDeleteFailed:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusCreateFailed:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusDeleteInProgress:
-		return false
-	default:
-		return false
+	status := ServiceStatus{
+		Arn: aws.StringValue(svcSum.Arn),
+		Id:  aws.StringValue(svcSum.Id),
+	}
+	if svcSum.DnsEntry != nil {
+		status.Dns = aws.StringValue(svcSum.DnsEntry.DomainName)
 	}
+	return status, nil
 }
 
-// find service by name return serviceNetwork,err if mesh exists, otherwise return nil,nil
-func (s *defaultServiceManager) findServiceByName(ctx context.Context, serviceName string) (*vpclattice.ServiceSummary, error) {
-	latticeSess := s.cloud.Lattice()
-	servicesListInput := vpclattice.ListServicesInput{}
-	resp, err := latticeSess.ListServicesAsList(ctx, &servicesListInput)
-	glog.V(6).Infof("findServiceByName, resp %v, err: %v\n", resp, err)
-
-	if err == nil {
-		for _, r := range resp {
-			if aws.StringValue(r.Name) == serviceName {
-				glog.V(6).Info("service ", serviceName, " already exists with arn ", *r.Arn, "\n")
-				return r, err
-			}
+// update SN-Svc associations, if svc has no SN associations will delete all of them
+// does not delete associations that are not tagged by controller
+func (m *defaultServiceManager) updateSnSvcAssocs(ctx context.Context, svc *Service, svcSum *SvcSummary) error {
```

Comment:
I'm not sure we should be using this method for both ```Create/Update``` and ```Delete```. On delete, shouldn't we be deleting all associations whether or not we created them?

Reply:
Actually I'm using this on Update and Delete but not create. But I agree that delete should be more straightforward with deletion. So create/update/delete will have own little method.